lib/vagrant/action/builtin/box_add.rb

```diff
@@ -478,13 +482,23 @@ def download(url, env, **opts)
             end
           end
 
+          mutex_path = d.destination + MUTEX_SUFFIX
+          if File.file?(mutex_path)
+            raise Errors::DownloadAlreadyInProgress,
+              dest_path: d.destination,
+              lock_file_path: mutex_path
+          end
+
           begin
+            File.write(mutex_path, "")
             d.download!
           rescue Errors::DownloaderInterrupted
             # The downloader was interrupted, so just return, because that
             # means we were interrupted as well.
             @download_interrupted = true
             env[:ui].info(I18n.t("vagrant.actions.box.download.interrupted"))
+          ensure
+            File.delete(mutex_path)
```

Comment:
I think this is an ok time to delete the mutex. I can imagine there being a situation between the download finishing and the box being added to the collection where another download of the same box starts causing an error. So, maybe this file needs to be created earlier and deleted earlier to better line up with the box being added to the box collection?